tuned/plugins/plugin_bootloader.py

```diff
@@ -13,10 +13,163 @@
 
 class BootloaderPlugin(base.Plugin):
 	"""
-	Plugin for tuning bootloader options.
-
-	Currently only grub2 is supported and reboot is required to apply the tunings.
-	These tunings are unloaded only on profile change followed by reboot.
+	`bootloader`::
+	
+	Adds options to the kernel command line. This plug-in supports the
+	GRUB 2 boot loader and the Boot Loader Specification (BLS).
+	+
+	NOTE: *TuneD* will not remove or replace kernel command line parameters
+	added via other methods like *grubby*. *TuneD* will only be able
+	manage kernel command line parameters added via *TuneD*. Please refer
+	to your platform bootloader documentation about how to identify and
+	manage kernel command line parameters set outside of *TuneD*.
+	+
+	Customized non-standard location of the GRUB 2 configuration file
+	can be specified by the [option]`grub2_cfg_file` option.
+	+
+	The kernel options are added to the current GRUB configuration and
+	its templates. The system needs to be rebooted for the kernel options
+	to take effect.
+	+
+	Switching to another profile or manually stopping the `tuned`
+	service removes the additional options. If you shut down or reboot
+	the system, the kernel options persist in the [filename]`grub.cfg`
+	file and grub environment files.
+	+
+	The kernel options can be specified by the following syntax:
+	+
+	[subs="+quotes,+macros"]
+	----
+	+cmdline+[replaceable]__suffix__=[replaceable]__arg1__ [replaceable]__arg2__ ... [replaceable]__argN__
+	----
+	+
+	Or with an alternative, but equivalent syntax:
+	+
+	[subs="+quotes,+macros"]
+	----
+	+cmdline+[replaceable]__suffix__=+[replaceable]__arg1__ [replaceable]__arg2__ ... [replaceable]__argN__
+	----
+	+
+	Where [replaceable]__suffix__ can be arbitrary (even empty) alphanumeric
+	string which should be unique across all loaded profiles. Good
+	practice is to use the profile name as the [replaceable]__suffix__
+	(e.g. [option]`cmdline_my_profile`). If there are multiple
+	[option]`cmdline` options with the same suffix (e.g. in the parent
+	profiles), then during the profile load/merge the value which was
+	assigned the last time wins (this is the same as with any other
+	plug-in options). Finaly, the kernel command line is constructed by
+	concatenation of the all resulting [option]`cmdline` options.
+	+
+	It is also possible to remove kernel options by the following syntax:
+	+
+	[subs="+quotes,+macros"]
+	----
+	+cmdline+[replaceable]__suffix__=-[replaceable]__arg1__ [replaceable]__arg2__ ... [replaceable]__argN__
+	----
+	+
+	Such kernel options will be removed (instead of concatenated) during
+	the final kernel command line construction.
+	+
```

Comment:
Such kernel options will not be concatenated and removed during the final kernel command line construction.